src/core/dev_api/openvino/core/descriptor_tensor.hpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/descriptor/tensor.hpp"
+
+namespace ov {
+namespace descriptor {
+
+// To change Tensor element type please change the Parameter type.
+void set_element_type(Tensor& tensor, const element::Type& elemenet_type);
+
+// To change Tensor type please change the Parameter type.
+void set_tensor_type(Tensor& tensor, const element::Type& element_type, const PartialShape& pshape);
+
+OPENVINO_DEPRECATED("get_ov_tensor_legacy_name() is deprecated. Please don't use this function.")
```

Comment:
@ilya-lavrenov 
Can we drop these functions if merged before 2024.0 release? 
Or in comment but release version when it will be removed?